source/parser.hera

```diff
@@ -1297,7 +1289,9 @@ BracedBlock
 
 # NOTE: SingleLineStatements includes the empty case
 SingleLineStatements
-  ( _* Statement SemicolonDelimiter )*:stmts ( _* Statement SemicolonDelimiter? )?:last ->
+  # NOTE: Statement can start with __ via AssignmentExpression.
+  # Force staying on the same line via !/\n/ assertion.
+  ( ( _? !/\n/ ) Statement SemicolonDelimiter )*:stmts ( ( _? !/\n/ ) Statement SemicolonDelimiter? )?:last ->
```

Comment:
I think `!EOS` is a better way to assert same line since it picks up `\r\n` as well.